Modules/Core/Common/src/itkPlatformMultiThreaderSingle.cxx

```diff
@@ -39,9 +39,9 @@ PlatformMultiThreader::MultipleMethodExecute()
   ThreadIdType thread_loop;
 
   // obey the global maximum number of threads limit
-  if (m_NumberOfWorkUnits > m_GlobalMaximumNumberOfThreads)
+  if (m_NumberOfWorkUnits > MultiThreaderBase::GetGlobalMaximumNumberOfThreads())
   {
-    m_NumberOfWorkUnits = m_GlobalMaximumNumberOfThreads;
+    m_NumberOfWorkUnits = MultiThreaderBase::GetGlobalMaximumNumberOfThreads();
```

Comment:
For my understanding, isn't the GlobalMaximumNumberOfThreads always exactly `1`, for MultiThreaderSingle?

Reply:
Yes, that is correct, itkThreadSupports.h sets

constexpr size_t ITK_MAX_THREADS = 1;

in case of non-pthreads/win32threads (line 78). However, the proposed change does not change the style compared to the other variants - and the 1 from ITK_MAX_THREADS is retrieved via header functions only (same performance).

Reply:
Ah, thanks for explaining, @fabian-philips! I see now, the same GetGlobalMaximumNumberOfThreads() check in itkPlatformMultiThreaderPosix.cxx and itkPlatformMultiThreaderWindows.cxx, indeed!